src/block/utils/FortuneTrait.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block\utils;
+
+use pocketmine\item\enchantment\VanillaEnchantments;
+use pocketmine\item\Item;
+use pocketmine\item\VanillaItems;
+use function array_merge;
+use function min;
+use function mt_getrandmax;
+use function mt_rand;
+use const PHP_INT_MAX;
+
+trait FortuneTrait{
+	/**
+	 * @return Item[]
+	 */
+	public function getDropsForCompatibleTool(Item $item) : array{
+		return $this->getFortuneDrops($item);
+	}
+
+	/**
+	 * @return Item[]
+	 */
+	public function getFortuneDrops(Item $item) : array{
+		$fortuneEnchantment = VanillaEnchantments::FORTUNE();
+		return $this->getFortuneDropsForLevel($item->getEnchantmentLevel($fortuneEnchantment));
+	}
+
+	/**
+	 * @return Item[]
+	 */
+	abstract protected function getFortuneDropsForLevel(int $level) : array;
+
+	/**
+	 * Gives a weight of 2 to a normal drop chance and adds a weight of 1 for each extra drop multiplier.
+	 *
+	 * @return Item[]
+	 */
+	protected function weightedDrops(Item $item, int $min, int $max, int $fortuneLevel) : array{
```

Comment:
A check for min < max should be added here, otherwise this will generate confusing errors on incorrect usage